### ISSUE TITLE
Deployment isn't working while no error appears but fleet-controller contains errors

### ISSUE THREAD

**Issue by Shaked** [rancher/fleet#222]
We have created a new deployment which we deploy to a specific ClusterGroup. 

Then, we have first tried to set `fleet.yaml` to override it with the following yaml:

```
namespace: development
helm:
  values:
    localDashboard: true
    image:
      repository: example/dashboard
rolloutStrategy:
  maxUnavailable: 100%
  maxUnavailablePartitions: 100%
targetCustomizations:
  - name: dashboard
    helm:
      values:
        hostNetwork: true
        mysql:
          image:
            repository: mysql/mysql-server
            tag: "8.0"
    clusterSelector:
      matchLabels:
        dashboard: true
        env: dev
``` 

However, the deployment showed `0` below the `Ready` clusters. 

We couldn't really figure what happened as no errors were shown and everything looked green. 

At some point, we checked  the logs in the `fleet-controller`pod, and we saw the following information: 

```
╰─$ ku fleet-system logs -f fleet-controller-56b7988677-8t9dn --since 10m | grep -i error -A 5 -B  5
time="2021-01-13T12:26:41Z" level=info msg="rate limited gitjobs(map[objectset.rio.cattle.io/hash:4f159f0a456ea7a85d02a8b3d0e8265552bdbf11]) 1.008463974s"
time="2021-01-13T12:26:41Z" level=info msg="rate limited gitjobs(map[objectset.rio.cattle.io/hash:32eba7225f2d45b7a3e93fa54a46e040401f43ce]) 1.003386356s"
time="2021-01-13T12:26:41Z" level=info msg="rate limited gitjobs(map[objectset.rio.cattle.io/hash:f3bdfb6bd012a6dee6ca35d77fbdbf76ef98c9de]) 1.004908432s"
time="2021-01-13T12:26:42Z" level=info msg="rate limited bundle(map[objectset.rio.cattle.io/hash:15385d232142edea68b36761ba872619360330a5]) 1.007646587s"
time="2021-01-13T12:26:43Z" level=info msg="rate limited gitjobs(map[objectset.rio.cattle.io/hash:3cecb004f649a37f8edb1bb61ad9dd7c2f9b3d70]) 1.004979031s"
time="2021-01-13T12:26:45Z" level=error msg="error syncing 'fleet-default/dashboard-dashboard': handler bundle: template: dashboard/templates/deployment.yaml:48:28: executing \"dashboard/templates/deployment.yaml\" at <.Values.mysql.image.repository>: nil pointer evaluating interface {}.image, requeuing"
time="2021-01-13T12:26:46Z" level=info msg="rate limited gitjobs(map[objectset.rio.cattle.io/hash:3ea3473f0716d950517ffd430433c1596c37f579]) 1.003835349s"
time="2021-01-13T12:26:49Z" level=info msg="rate limited gitjobs(map[objectset.rio.cattle.io/hash:3ea3473f0716d950517ffd430433c1596c37f579]) 1.001197792s"
time="2021-01-13T12:26:50Z" level=info msg="rate limited gitjobs(map[objectset.rio.cattle.io/hash:437839a8c7478db68d09786f0ddffa4bcd1102e8]) 1.004728434s"
```

We figured that the default values for our deployment were empty, so we tried to change our `fleet.yaml` to the following yaml: 

```
namespace: development
helm:
  values:
    localDashboard: true
    image:
      repository: example/dashboard
    mysql:
      image:
        repository: mysql/mysql-server
        tag: "8.0"
rolloutStrategy:
  maxUnavailable: 100%
  maxUnavailablePartitions: 100%
targetCustomizations:
  - name: dashboard
    helm:
      values:
        hostNetwork: true
    clusterSelector:
      matchLabels:
        dashboard: true
        env: dev
```

Moving the `mysql` values to the default helm values. 

Once we pushed it, everything started to work. 

We would expected the errors to be shown in rancher's dashboard, telling us what we did wrong as it took some time to figure. 

Also, there are still lots of `level=info msg="rate limited gitjobs` messages going on in the logs, could it be related? We saw this errors also in rancher/fleet#216 

Thanks, 
Shaked

**Comment by kkaempf** [comment]
to be moved to rancher/dashboard

**Comment by kkaempf** [comment]
fixed by #1123
